src/chainrules.jl

```diff
@@ -6,3 +6,122 @@ ChainRulesCore.@scalar_rule(
     ),
     (x, - tanh(Ω + b) * x, x - 1),
 )
+
+# `OrderedBijector`
+function ChainRulesCore.rrule(::typeof(_transform_ordered), y::AbstractVector)
+    function _transform_ordered_adjoint(Δ)
+        Δ_new = similar(y)
+        n = length(Δ)
+        @assert n == length(Δ_new)
+
+        s = sum(Δ)
+        Δ_new[1] = s
+        @inbounds for i in 2:n
+            # Equivalent to
+            #
+            #    Δ_new[i] = sum(Δ[i:end]) * yexp[i - 1]
+            #
+            s -= Δ[i - 1]
+            Δ_new[i] = s * exp(y[i])
+        end
+
+        return (ChainRulesCore.NoTangent(), Δ_new)
+    end
+
+    return _transform_ordered(y), _transform_ordered_adjoint
+end
+
+function ChainRulesCore.rrule(::typeof(_transform_ordered), y::AbstractMatrix)
+    function _transform_ordered_adjoint(Δ)
+        Δ_new = similar(y)
+        n = length(Δ)
+        @assert n == size(Δ_new, 1)
+
+        s = sum(Δ)
+        Δ_new[1, :] .= s
+        @inbounds for i in 2:n
+            # Equivalent to
+            #
+            #    Δ_new[i] = sum(Δ[i:end]) * yexp[i - 1]
+            #
+            s -= Δ[i - 1]
+            Δ_new[i, :] = s * exp.(y[i, :])
+        end
+
+        return (ChainRulesCore.NoTangent(), Δ_new)
+    end
+
+    return _transform_ordered(y), _transform_ordered_adjoint
+end
+
+function ChainRulesCore.rrule(::typeof(_transform_inverse_ordered), x::AbstractVector)
+    r = similar(x)
+    @inbounds for i = 1:length(r)
+        if i == 1
+            r[i] = 1
+        else
+            r[i] = x[i] - x[i - 1]
+        end
+    end
+
+    function _transform_inverse_ordered_adjoint(Δ)
+        Δ_new = similar(x)
+        @assert length(Δ_new) == length(Δ)
+
+        n = length(Δ_new)
+        @inbounds for j = 1:n - 1
+            Δ_new[j] = (Δ[j] / r[j]) - (Δ[j + 1] / r[j + 1])
```

Comment:
`Δ[j] / r[j]` is already computed in the previous step. Maybe save it?

Reply:
True, but it requires a couple of more variables. A quick benchmark shows that it doesn't do anything for perf :confused: 
Could this be because of cache prefetching, i.e. the adjacent variables are already on the CPU and so it doesn't matter.